Light drifter button ai

AkashCherukuri opened the pull request.

AkashCherukuri committed:
Add new xmlrpc function to accept button request

A new XMLRPC function called add has been added which takes the request from the Add Bot Button

AkashCherukuri committed:
Add new Function for AddBot Button

AkashCherukuri committed:
Add new AddBot Button

Added a new button called Add bot in a similar fashion of Start Game

AkashCherukuri committed:
Add Button Position

AkashCherukuri committed:
Debug Hide Button when game starts

AkashCherukuri committed:
Modify AddBot() to set Bot Alias Cookies

AkashCherukuri committed:
Add functionality to be able to play with bots

cheekujodhpur commented on game/core.py:
```diff
@@ -63,6 +63,7 @@ def add_player(self, alias):
 
     def add_bot(self):
         if len(self.players) < 6:
+            self.num_bots+=1
```
Why do you need this self.num_bots variable? You can create a function which counts all players with AI flag True

AkashCherukuri replied:
Okay, yeah I've gone ahead and done that

cheekujodhpur commented on game/core.py:
```diff
@@ -311,6 +314,10 @@ def xmlrpc_query_state(self, request, game_id, player_token):
             result["hand"] = player.hand
             result["top_card"], result["top_card_v"] = game.deck.top_card()
+
+            if(game.turn.token[0].islower()):
```
more pythonic syntax please. `if game.turn.token[0].islower():`

AkashCherukuri replied:
Did that, will be more "pythonic" from next time

AkashCherukuri committed:
Fix bug to end round when no player is active

The added integers enable usage of sum() in core.py

AkashCherukuri committed:
Fix but to end round when no player is active

sum() is applicable on act defined in players.py

AkashCherukuri committed:
Debug Hide button

AkashCherukuri committed:
Debug scoring of bots and occasional game crashes

AkashCherukuri committed:
Debug logging to be in correct order

AkashCherukuri committed:
Removed unnecessary Variables

AkashCherukuri committed:
Debug incorrect Round ending

The sum() function present previously works only on integers. This causes the game to end if all the bots have folded OR sometimes the round doesn't end even if all the players have folded.

AkashCherukuri committed:
Add bool isBot 

This boolean can be used to know whether the player is a bot or not, as the player token may or may not be available to us.

AkashCherukuri committed:
Add function to change isbot var

AkashCherukuri committed:
Merge branch 'AI-Single' into LightDrifter-Button-Ai
